augur/filter/subsample.py

```diff
@@ -124,11 +124,11 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
             # to generate other columns, and will be discarded at the end.
             temp_prefix = str(uuid.uuid4())
             temp_date_cols = [f'{temp_prefix}year', f'{temp_prefix}month', f'{temp_prefix}day']
-            df_dates = metadata[METADATA_DATE_COLUMN].str.split('-', n=2, expand=True)
-            df_dates = df_dates.set_axis(temp_date_cols[:len(df_dates.columns)], axis=1)
-            missing_date_cols = set(temp_date_cols) - set(df_dates.columns)
-            for col in missing_date_cols:
-                df_dates[col] = pd.NA
+            df_dates = pd.DataFrame(
+                metadata[METADATA_DATE_COLUMN].apply(get_year_month_day).tolist(),
+                columns=temp_date_cols,
+                index=metadata.index,
+            )
```

Comment:
This could negatively impact run times, especially for large metadata files. It's released now, so I'll keep an eye out for the next ncov workflow run and check the run time difference there. We could add [`@cache`](https://docs.python.org/3/library/functools.html#functools.cache) to one of the underlying date parsing functions to improve run times, since dates should be highly repetitive in large datasets.

Reply:
So far ([n=2](https://bedfordlab.slack.com/archives/CSKMU6YUC/p1744826671189259?thread_ts=1744655123.351749&cid=CSKMU6YUC)) there has been no noticeable difference in run times, but I did some quick testing and `@cache` seems to work well. I'll make a separate PR.

Reply:
#1792